wandb/sdk/wandb_run.py

```diff
@@ -314,6 +379,106 @@ def _display_run(self):
             )
         )
 
+    def _redirect(self, stdout_slave_fd, stderr_slave_fd):
+        console = self._settings.console
+        logger.info("redirect: %s", console)
+
+        if console == "redirect":
+            logger.info("redirect1")
+            out_cap = redirect.Capture(name="stdout", cb=self._redirect_cb)
+            out_redir = redirect.Redirect(
+                src="stdout", dest=out_cap, unbuffered=True, tee=True
+            )
+            err_cap = redirect.Capture(name="stderr", cb=self._redirect_cb)
+            err_redir = redirect.Redirect(
+                src="stderr", dest=err_cap, unbuffered=True, tee=True
+            )
+            try:
+                out_redir.install()
+                err_redir.install()
+                self._out_redir = out_redir
+                self._err_redir = err_redir
+                logger.info("redirect2")
+            except (OSError, AttributeError) as e:
+                logger.error("failed to redirect", exc_info=e)
+            return
+
+        return
+
+        # redirect stdout
+        if platform.system() == "Windows":
+            lib_console.win32_redirect(stdout_slave_fd, stderr_slave_fd)
```

Comment:
This is a noop currently right?  We may want a comment here or something to indicate what's happening in this case.

Reply:
fixed